### ISSUE TITLE
Conflict with user roles and ACF relationship field

### ISSUE THREAD

**Issue by phylaxis** [woocommerce/woocommerce#14580]
### EXPLANATION OF THE ISSUE

We have a site where we're using a ACF Pro Relationship field pulling users and filtering based on roles. The filed is used to add "additional" authors to our posts. The issue we are seeing is that when WooCommerce 3.0 (and earlier) and 'Administrator' role users are being filtered out of the list. We tested this running the TwentySeventeen theme and with only WC 3.0 and ACF Pro 5.5.11 installed. Removing WC from that equation filed the issue.

We began discussing the conflict with ACF support in this thread and there are additional details there: https://support.advancedcustomfields.com/forums/topic/relationship-user-field-cant-see-administrator-users/

Basically we THINK that there is some filter in Woo Commerce adjusting the user roles before ACF gets them. We tried filtering out the `wc_modify_editable_roles` for the one filed in ACF we're having issues with but that didn't seem to be it. Is there possibly another place/filter we should look at? 

### STEPS TO REPRODUCE THE ISSUE

1. Install WC 3.0 and ACF Pro 5.5.11 with the TwentySeventeen theme

2. Set up an ACF Relationship field for Users and set filter to return users with Administrator role

3. Log in as a non-Administrator role with post permissions (Author, Editor, etc).

4. Create a new post and try to return an Admin user from the ACF relationship field. No results are displayed.

### SYSTEM STATUS

<details>
```
### WordPress Environment ###

Home URL: http://wp.macobserver.com
Site URL: http://wp.macobserver.com
WC Version: 3.0.3
Log Directory Writable: ✔
WP Version: 4.7.3
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US

### Server Environment ###

Server Info: Apache/2.4.25 (Unix) LibreSSL/2.2.7 PHP/5.6.30
PHP Version: 5.6.30
PHP Post Max Size: 8 MB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 7.51.0
SecureTransport

SUHOSIN Installed: –
MySQL Version: 5.7.17
Max Upload Size: 2 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.0.3
WC Database Prefix: wp_
woocommerce_sessions: ✔
woocommerce_api_keys: ✔
woocommerce_attribute_taxonomies: ✔
woocommerce_downloadable_product_permissions: ✔
woocommerce_order_items: ✔
woocommerce_order_itemmeta: ✔
woocommerce_tax_rates: ✔
woocommerce_tax_rate_locations: ✔
woocommerce_shipping_zones: ✔
woocommerce_shipping_zone_locations: ✔
woocommerce_shipping_zone_methods: ✔
woocommerce_payment_tokens: ✔
woocommerce_payment_tokenmeta: ✔
MaxMind GeoIP Database: ✔

### Security ###

Secure connection (HTTPS): ❌Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.
Hide errors from visitors: ❌Error messages should not be shown to visitors.

### Active Plugins (2) ###

Advanced Custom Fields PRO: by Elliot Condon – 5.5.11
WooCommerce: by Automattic – 3.0.3

### Settings ###

API Enabled: ✔
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
subscription (subscription)
variable (variable)
variable subscription (variable-subscription)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)


### WC Pages ###

Shop base: #363 - /shop/
Cart: #364 - /cart/
Checkout: #365 - /checkout/
My account: #366 - /mgg-account/

### Theme ###

Name: Twenty Seventeen
Version: 1.1
Author URL: https://wordpress.org/
Child Theme: ❌ – If you're modifying WooCommerce on a parent theme you didn't build personally
then we recommend using a child theme. See: How to create a child theme

WooCommerce Support: ✔

### Templates ###

Overrides: –

```
</details>

**Comment by mikejolley** [comment]
There is indeed some code which looks to be security related:

```
/**
 * Modify the list of editable roles to prevent non-admin adding admin users.
 * @param  array $roles
 * @return array
 */
function wc_modify_editable_roles( $roles ) {
	if ( ! current_user_can( 'administrator' ) ) {
		unset( $roles['administrator'] );
	}
	return $roles;
}
add_filter( 'editable_roles', 'wc_modify_editable_roles' );
```

The code they gave you will work. We'll need to review why this is present before taking any action.

**Comment by phylaxis** [comment]
Wow. Quick reply. Thanks!

**Comment by mikejolley** [comment]
@phylaxis Done some research and we need to keep this code around (you can disable with the code you were provided).

This is the issue the code we have resolves: https://github.com/woocommerce/woocommerce/issues/5322

This is the WordPress trac ticket explaining the issue: https://core.trac.wordpress.org/ticket/14986

TL;DR shop_managers have edit_users capability, which allows them to upgrade any user to admin without our code in place.

Hope that makes sense.

**Comment by phylaxis** [comment]
Ok. Thanks for looking into this @mikejolley.
